CRM/Member/Form/MembershipRenewal.php

```diff
@@ -142,7 +144,6 @@ public function preProcess() {
 
     // @todo - we should store this as a property & re-use in setDefaults - for now that's a bigger change.
     $currentMembership = civicrm_api3('Membership', 'getsingle', ['id' => $this->_id]);
-    CRM_Member_BAO_Membership::fixMembershipStatusBeforeRenew($currentMembership);
```

Comment:
note that there was a reason this was put here - https://github.com/civicrm/civicrm-core/commit/094ff61f986a292c8acfc69f45aabdb28eb58c39 (I haven't re-read it though)